index.js

```diff
@@ -117,6 +118,10 @@ Shopify.prototype.request = function request(url, method, key, params) {
     options.headers['X-Shopify-Access-Token'] = this.options.accessToken;
   }
 
+  if (this.options.presentmentPrices) {
+    options.headers['X-Shopify-Api-Features'] = 'include-presentment-prices';
```

Comment:
I would prefer this header to be only set for the product and productVariant resources.

Reply:
Done

As a side note, I think the conditional logic is rather bloated. If you have any clout with the Shopify folks, we might try to persuade them to make this an option in the individual requests. Requiring this as a header is outside the norm for their API's and annoying to handle consistently.

Reply:
My idea was to actually add the header in the implementation of the relevant resources. For example overriding `buildUrl()` in https://github.com/MONEI/Shopify-api-node/blob/master/resources/product.js and adding the header if the option is set.

We can then set the `User-Agent` and `X-Shopify-Access-Token` headers after the options object is built. Does it make sense?

Reply:
I was thinking something similar, but was looking at the functions in the base-child mixin. I like your proposal but it's unclear to me how the overriding works in this scenario. If I create a BuildUrl field on the product prototype, won't it be overwritten with the BuildUrl in base when we call assign(Product.Prototype, base)?

The other concern was with visibility about the headers being set. Everything for the requests is being handled in that one function right now. If we start setting headers inside the resources, it won't be immediately obvious. This could get confusing and potentially lead to bugs later if someone adds a default header to the option object due to a change in the APIs. That default header will get overridden when these resources are used.

Reply:
Yes I recognise that current design does not allow to do this in a clean way but I don't like this regex based approach. If there is no other way I would prefer to go back, revert eeb6bcb, and set it on every request.

Reply:
I've addressed the feedback by overloading the base functions where required in the resources. I have concerns about the headers being shadowed later with how the assign function works, but I suppose unit testing will have to serve as a defense for now.